dartsim/src/SDFFeatures.cc

```diff
@@ -600,6 +600,31 @@ Identity SDFFeatures::ConstructSdfLink(
 
   bodyProperties.mInertia.setLocalCOM(localCom);
 
+  // If added mass is provided, add that to DART's computed spatial tensor
+  // TODO(chapulina) Put in another feature
+  if (sdfInertia.FluidAddedMass().has_value())
+  {
+    // TODO check off-diagonal blocks
+    math::Matrix6d featherstoneMatrix;
+    featherstoneMatrix.SetSubmatrix(math::Matrix6d::TOP_LEFT,
+        sdfInertia.FluidAddedMass().value().Submatrix(
+        math::Matrix6d::BOTTOM_RIGHT));
```

Comment:
This appears to be missing the term `mMass*C*C.transpose()` that appears in the Dart spatial tensor:

https://github.com/dartsim/dart/blob/26267f384cd0e5f4c00a6b074005902313df855b/dart/dynamics/Inertia.cpp#L393-L398

```c++
void Inertia::computeSpatialTensor()
{
  Eigen::Matrix3d C = math::makeSkewSymmetric(mCenterOfMass);

// Top left
mSpatialTensor.block<3,3>(0,0) = getMoment() + mMass*C*C.transpose();
```

Reply:
I've created a patch in the commits below - the first includes debug statements which are cleaned up in the second.

https://github.com/srmainwaring/gz-physics/commit/9dba6cd30f14a017a13e01b6088c92071b5ea2ad
https://github.com/srmainwaring/gz-physics/commit/b60be398a990684da092bf7e3f4eec1f296058ce

```diff
+
+    // Calculate CoM term.
+    //
+    // Roy Featherstone, Rigid Body Dynamics Algorithms, Springer
+    // The spatial inertia matrix is defined in §2.13, p33, eq(2.63).
+    auto M = sdfInertia.FluidAddedMass().value().Submatrix(
+        math::Matrix6d::TOP_LEFT);
+
+    auto CM = sdfInertia.FluidAddedMass().value().Submatrix(
+        math::Matrix6d::BOTTOM_LEFT);
+
+    auto invM = M.Inverse();
+    auto C = CM * invM;
+    auto CMCT = CM * C.Transposed();
+
+    // I + C * M * C.transpose
     featherstoneMatrix.SetSubmatrix(math::Matrix6d::TOP_LEFT,
         sdfInertia.FluidAddedMass().value().Submatrix(
-        math::Matrix6d::BOTTOM_RIGHT));
+        math::Matrix6d::BOTTOM_RIGHT) + CMCT);
```

Reply:
Right now, we are only allowing added mass to be specified in the link frame.  These computations would be required for translating to the center of mass.  For now, this is unnecessary.